contents/code/main.js

```diff
@@ -1,143 +1,136 @@
-}
 
-function is_last_empty()
-{
-	// -1 as we always have the last desktop that we keep empty for moving
-	// things to it
-	return is_empty_desktop(workspace.desktops - 1);
-}
-
-function delete_empty_last()
-{
-	if (is_last_empty()) {
-		delete_last();
-		print("deleted last desktop");
-	}
+	return true;
 }
 
+/**
+ * Checks for new created or moved windows if they are occupying the last desktop
+ * -> if yes, create new one to the right
+ */
 function desktop_changed_for(client)
 {
-	return function() {
-		var message = "Client " + client.caption + " just moved";
-		message += "\n to desktop number " + client.desktop;
-		message += " out of " + workspace.desktops;
-
-		if (client.desktop >= workspace.desktops) {
-			add_desktop();
-			message += "\nadded a desktop";
-		}
-		else {
-			delete_empty_last();
-		}
+	print(`desktop_changed_for() -> Client ${client.caption} just moved to desktop number ${client.desktop}`);
 
-		print(message);
+	if (client.desktop >= workspace.desktops)
+	{
+		add_desktop();
 	}
 }
 
-
+/**
+ * When creating new windows, check whether they are occupying the last desktop
+ */
 function on_client_added(client)
 {
-	if (client === null) {
+	if (client === null)
+	{
 		// just in case
 		return;
 	}
 
-	if (client.skipPager) {
+	if (client.skipPager)
+	{
 		//ignore hidden windows
 		return;
 	}
 
 	// add a new desktop for a client too right
-	if (client.desktop >= workspace.desktops) {
+	if (client.desktop >= workspace.desktops)
+	{
 		add_desktop();
 	}
 
 	// subscribe the client to create desktops when desktop switched
-	client.desktopChanged.connect(desktop_changed_for(client))
+	client.desktopChanged.connect(() => { desktop_changed_for(client); });
 }
 
-function on_desktop_changed(old_desktop, client)
+/**
+ * Deletes empty desktops to the right in case of a left switch
+ */
+function on_desktop_switch(old_desktop)
 {
-	// delete empty desktops that we swithced from
+	print(`on_desktop_switch(${old_desktop})`);
 
-	if (old_desktop !== workspace.desktops && is_empty_desktop(old_desktop)) {
-		// delete desktop
-		// only delete desktop if doing so would be unnoticeable
-		if (old_desktop > workspace.currentDesktop) {
-			delete_desktop(old_desktop);
-		}
-	} else if (workspace.currentDesktop === 1) {
-		// delete all empty desktops to the right if we switched to first
-		for (var i = 1; i < workspace.desktops; ++i) {
-			if (is_empty_desktop(i)) {
-				var deleted = delete_desktop(i);
-				if (deleted) {
-					i -= 1;
-				}
-			}
+	// do nothing if we switched to the right
+	if (old_desktop <= workspace.currentDesktop) return;
+
+	// start from next desktop to the right
+	let desktop = workspace.currentDesktop + 1;
+
+	// prevent infinit loop in case of an error
+	// might happen if other plugins interfere with workspace creation/deletion
+	let loop_counter = 0;
+	let loop_limit = workspace.desktops;
+	for (; desktop < workspace.desktops && loop_counter < loop_limit; desktop++)
+	{
+		loop_counter++;
```

Comment:
Isn't it possible in JS to do as in C and put this in the last statement of for loop as well? like `(..; desktop++, loop_counter++)`?

Reply:
Jup you are right, this also works